python/triton/knobs.py

```diff
@@ -52,16 +81,21 @@ def __get__(self, obj: Optional[object], objclass: Optional[Type[object]]) -> Ge
         else:
             return self.get()
 
+    @property
+    def env_val(self) -> str | None:
+        return getenv(self.key)
+
     def get(self) -> GetType:
-        env = getenv(self.key)
+        env = self.env_val
         return self.transform(self.default() if env is None else self.from_env(env))
 
     def __set__(self, obj: object, value: Union[SetType, Env]) -> None:
         if isinstance(value, Env):
             obj.__dict__.pop(self.name, None)
         else:
             obj.__dict__[self.name] = value
-            self.set(value)
+            if env_val := toenv(value):
+                setenv(self.key, env_val[0])
```

Comment:
Almost exactly the same impl you gave @peterbell10 , but needed a function to convert the value (e.g. "1" instead of "True" and to ignore when e.g. value is a class for `env_class`)

Reply:
What is the tuple for?

Reply:
To differentiate the case where the env var should be unset from when the environment shouldn't be updated-- I'm using the tuple as a sort of 'box', e.g. I'd like the retval to be:
```
Box(str | None) | None
```

Reply:
An alternative I played with was to introduce another singleton 'Unset` instead of the `None` inside the box, but figured we have complex enough types, maybe just using the tuple would be fine.

I stopped the automerge for now so that we can change this up as you see fit